src/Android/Xamarin.Android/Samples/GeometryEngine/ClipGeometry/ClipGeometry.cs

```diff
@@ -0,0 +1,258 @@
+
+            // Create the graphic for Colorado - comprised of a polygon shape and fill symbol.
+            _coloradoGraphic = new Graphic(colorado, coloradoSimpleFillSymbol);
+
+            // Add the Colorado graphic to the input geometries graphics overlay collection.
+            _inputGeometriesGraphicsOverlay.Graphics.Add(_coloradoGraphic);
+
+            // Create a simple line symbol for the three different clip geometries.
+            SimpleLineSymbol clipGeomtriesSimpleLineSymbol = new SimpleLineSymbol(
+                SimpleLineSymbolStyle.Dot, System.Drawing.Color.Red, 5);
+
+            // Create an envelope outside Colorado.
+            Envelope outsideEnvelope = new Envelope(
+                new MapPoint(-11858344.321294, 5147942.225174, SpatialReferences.WebMercator),
+                new MapPoint(-12201990.219681, 5297071.577304, SpatialReferences.WebMercator));
+
+            // Create the graphic for an envelope outside Colorado - comprised of a polyline shape and line symbol.
+            _outsideGraphic = new Graphic(outsideEnvelope, clipGeomtriesSimpleLineSymbol);
+
+            // Add the envelope outside Colorado graphic to the graphics overlay collection.
+            _inputGeometriesGraphicsOverlay.Graphics.Add(_outsideGraphic);
+
+            // Create an envelope intersecting Colorado.
+            Envelope intersectingEnvelope = new Envelope(
+                new MapPoint(-11962086.479298, 4566553.881363, SpatialReferences.WebMercator),
+                new MapPoint(-12260345.183558, 4332053.378376, SpatialReferences.WebMercator));
+
+            // Create the graphic for an envelope intersecting Colorado - comprised of a polyline shape and line symbol.
+            _intersectingGraphic = new Graphic(intersectingEnvelope, clipGeomtriesSimpleLineSymbol);
+
+            // Add the envelope intersecting Colorado graphic to the graphics overlay collection.
+            _inputGeometriesGraphicsOverlay.Graphics.Add(_intersectingGraphic);
+
+            // Create a envelope inside Colorado.
+            Envelope containedEnvelope = new Envelope(
+                new MapPoint(-11655182.595204, 4741618.772994, SpatialReferences.WebMercator),
+                new MapPoint(-11431488.567009, 4593570.068343, SpatialReferences.WebMercator));
+
+            // Create the graphic for an envelope inside Colorado - comprised of a polyline shape and line symbol.
+            _containedGraphic = new Graphic(containedEnvelope, clipGeomtriesSimpleLineSymbol);
+
+            // Add the envelop inside Colorado graphic to the graphics overlay collection.
+            _inputGeometriesGraphicsOverlay.Graphics.Add(_containedGraphic);
+
+            // Get the extent of all of the graphics in the graphics overlay with a little padding to used as the initial zoom extent of the map.
+            Geometry visibleExtent = GetExtentOfGraphicsOverlay(_inputGeometriesGraphicsOverlay, 1.3, SpatialReferences.WebMercator);
```

Comment:
We could also do this with the `Extent` property of the graphics overlay, but we'd need to handle the `SpatialReferenceChanged` event to make sure it's initialized:
```
            MyMapView.SpatialReferenceChanged += (s, e) =>
            {
                EnvelopeBuilder extentEnvelopeBuilder = new EnvelopeBuilder(_inputGeometriesGraphicsOverlay.Extent);
                extentEnvelopeBuilder.Expand(1.3);
                Geometry visibleExtent = extentEnvelopeBuilder.ToGeometry();
                MyMapView.SetViewpointGeometryAsync(visibleExtent);
            };
```